### ISSUE TITLE
Using orderBy inside ManyToMany relationship causes an error

### ISSUE THREAD

**Issue by john-psina** [MrThearMan/graphene-django-query-optimizer#143]
### I have read the docs thoroughly before making this bug report.

- [x] Yes

### I have read through other open issues, and my issue is not a duplicate.

- [X] Yes

### What version of the library you are using?

0.10.0

### Which python version are you using?

3.12

### What operating system are you on?

Windows

### Description

The way to reproduce the error:

models.py:
```models.py
class Category(MPTTModel):
    parent = TreeForeignKey('self', verbose_name=_('admin__parent'), null=True, blank=True, related_name='children', on_delete=models.CASCADE)
    name = models.CharField(verbose_name=_('admin__name'), null=False, max_length=255)
    
    ...
    
    class Meta:
        verbose_name = _('admin__category')
        verbose_name_plural = _('admin__categories')
        ordering = ['tree_id', 'lft']
    
class Product(models.Model):
    show_in_categories = models.ManyToManyField(Category, verbose_name=_('admin__show_in_categories'), blank=True, )
    name = models.CharField(verbose_name=_('admin__name'), null=False, max_length=255)
    
    class Meta:
        verbose_name = _('admin__product')
        verbose_name_plural = _('admin__products')
        ordering = ['id',]
```

query.py:
```query.py
class CategoryFilter(ModelFilterSet):
    filter = UserDefinedFilter(
        model=Category,
        fields=['name', 'id'],
    )

    class Meta:
        order_by = ['id', 'name', 'parent', ]
        model = Category

class ProductFilter(ModelFilterSet):
    filter = UserDefinedFilter(
        model=Product,
        fields=[
            'name',
        ],
    )

    class Meta:
        order_by = ['id', 'name', ]
        model = Product
        
        
class CategoryType(DjangoObjectTypeLanguage):
    product_set = DjangoConnectionField(lambda: ProductType)
    children = graphene.List(lambda: CategoryType)

    class Meta:
        model = Category
        interfaces = (graphene.relay.Node,)
        fields = (
            'id',
            'name',
            'parent',
            'product_set',
            'level',
        )
        filterset_class = CategoryFilter
        
    def resolve_product_set(self, info, **kwargs):
        return Product.objects.calculate_price().filter(show_in_categories=self.id)
        
class ProductType(DjangoObjectTypeLanguage):
    show_in_categories = DjangoConnectionField(CategoryType)

    class Meta:
        model = Product
        interfaces = (graphene.relay.Node,)
        fields = [
            'id',
            'name',
            'show_in_categories',
        ]
        filterset_class = ProductFilter
        
        
class ProductQuery(graphene.ObjectType):
    categories = DjangoConnectionField(CategoryType)
    products = DjangoConnectionField(ProductType)
    
    def resolve_categories(self, info, **kwargs):
        return Category.objects.all()
        
    def resolve_products(self, info, **kwargs):
        return Product.objects.all()
    
````

My query without any errors:
```gql
query QueryWithoutErrors{
  categories(orderBy: nameAsc) {
    edges {
      node {
        name
        productSet {
          edges {
            node {
              name
            }
          }
        }
      }
    }
  }
}
```

Query that causes errors:
```gql
query QueryThatCausesError{
  categories {
    edges {
      node {
        name
        productSet(orderBy: nameAsc) {
          edges {
            node {
              name
            }
          }
        }
      }
    }
  }
}
```
response:
```json
{
  "errors": [
    {
      "message": "'list' object has no attribute 'split'",
      "locations": [
        {
          "line": 2,
          "column": 3
        }
      ],
      "path": [
        "categories"
      ],
      "extensions": {
        "type": "AttributeError",
        "code": "error",
        "timestamp": 1725382039,
        "operation": "QUERY",
        "trace": [
          "  File \"D:\\project\\venv1\\Lib\\site-packages\\graphql\\execution\\execute.py\", line 521, in execute_field\n    result = resolve_fn(source, info, **args)\n             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
          "  File \"D:\\project\\unine_engine\\middleware\\GraphqlDebugMiddleware.py\", line 20, in resolve\n    result = next(root, info, **args)\n             ^^^^^^^^^^^^^^^^^^^^^^^^\n",
          "  File \"D:\\project\\graphql_jwt\\middleware.py\", line 80, in resolve\n    return next(root, info, **kwargs)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
          "  File \"D:\\project\\venv1\\Lib\\site-packages\\query_optimizer\\fields.py\", line 295, in connection_resolver\n    queryset = optimizer.optimize_queryset(queryset)\n               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
          "  File \"D:\\project\\venv1\\Lib\\site-packages\\query_optimizer\\optimizer.py\", line 106, in optimize_queryset\n    results = self.process(queryset, filter_info)\n              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
          "  File \"D:\\project\\venv1\\Lib\\site-packages\\query_optimizer\\optimizer.py\", line 153, in process\n    prefetch = optimizer.process_prefetch(name, nested_results, nested_filter_info)\n               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
          "  File \"D:\\project\\venv1\\Lib\\site-packages\\query_optimizer\\optimizer.py\", line 181, in process_prefetch\n    queryset = self.paginate_prefetch_queryset(queryset, filter_info)\n               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
          "  File \"D:\\project\\venv1\\Lib\\site-packages\\query_optimizer\\optimizer.py\", line 204, in paginate_prefetch_queryset\n    [x for x in filter_info.get(\"filters\", {}).get(\"order_by\", \"\").split(\",\") if x]\n                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n"
        ]
      }
    }
  ]
  }
```

The error occurs in the file "optimizer.py" in the function "paginate_prefetch_queryset".

The query_optimizer code expects a string inside filter_info["filters"]["order_by"], but through the debugger I saw that the filter_info is:
```json
{
   "filters":{
      "order_by":[
         "<ProductOrderingChoices.nameAsc: 'name'>"
      ]
   },
   "filterset_class":"<class 'apps.products.graphql.query.ProductFilter'>"
}
```

The error disappears if this code line:
`[x for x in filter_info.get("filters", {}).get("order_by", "").split(",") if x]`
Replace with:
`[x for x in filter_info.get("filters", {}).get("order_by", "") if x]`

But I'm not sure if this is the right way to solve this problem, and if it will cause any other errors. I personally have not found any errors with the modified code, but I can't be sure that they won't happen.

**Comment by john-psina** [comment]
UPD:
If you fully comment:
```py
 order_by: list[str] = (
 # Use the `order_by` from the filter info, if available
 [x for x in filter_info.get("filters", {}).get("order_by", "").split(",") if x]
 # Use the model's `Meta.ordering` if no `order_by` is given
 or copy(queryset.model._meta.ordering)
 # No ordering if neither is available
 or []
 )
```
The error will disappear and order_by will still work correctly

**Comment by john-psina** [comment]
#144

**Comment by MrThearMan** [comment]
Hi, thanks for the issue.

From looking into this, seems like this is not an issue with this library, but [graphene-django-extensions](https://github.com/MrThearMan/graphene-django-extensions) (another one of mine) which you seem to be using. It has a slightly different ordering mechanism, which seems to conflict with the default one from `graphene-django`.

I'll add support for this difference, since its my library afterall.

**Comment by MrThearMan** [comment]
Made a change that should fix the compatibility issue. Released in [0.10.1](https://github.com/MrThearMan/graphene-django-query-optimizer/releases/tag/v0.10.1).

**Comment by john-psina** [comment]
Thank you so much!
